src/Client/Updater.php

```diff
@@ -518,28 +518,26 @@ protected function getMetadataForTarget(string $target, ?TargetsMetadata $target
             }
 
             $this->signatureVerifier->addRole($delegatedRole);
-            if (!$delegatedRole->matchesPath($target)) {
-                // Targets must match the path in all roles in the delegation chain so if the path does not match
+            if ($delegatedRole->matchesPath($target)) {
+                // Targets must match the path in all roles in the delegation chain, so if the path does not match,
                 // do not evaluate this role or any roles it delegates to.
-                continue;
-            }
-
-            $this->fetchAndVerifyTargetsMetadata($delegatedRoleName);
-            /** @var \Tuf\Metadata\TargetsMetadata $newTargetsData */
-            $newTargetsData = $this->metadataFactory->load($delegatedRoleName);
-            if ($newTargetsData->hasTarget($target)) {
-                return $newTargetsData;
-            }
-            $searchedRoles[] = $delegatedRole;
-            // § 5.6.7.2.1
-            //  If the current delegation is a multi-role delegation, recursively visit each role, and check that each has signed exactly the same non-custom metadata (i.e., length and hashes) about the target (or the lack of any such metadata).
-            if ($matchingTargetMetadata = $this->getMetadataForTarget($target, $newTargetsData, $searchedRoles)) {
-                return $matchingTargetMetadata;
-            }
-            if ($delegatedRole->isTerminating()) {
-                // § 5.6.7.2.2
-                // If the role is terminating then abort searching for a target.
-                return null;
+                $this->fetchAndVerifyTargetsMetadata($delegatedRoleName);
+                /** @var \Tuf\Metadata\TargetsMetadata $newTargetsData */
+                $newTargetsData = $this->metadataFactory->load($delegatedRoleName);
+                if ($newTargetsData->hasTarget($target)) {
+                    return $newTargetsData;
+                }
+                $searchedRoles[] = $delegatedRoleName;
+                // § 5.6.7.2.1
+                //  If the current delegation is a multi-role delegation, recursively visit each role, and check that each has signed exactly the same non-custom metadata (i.e., length and hashes) about the target (or the lack of any such metadata).
```

Comment:
Question: how do we know if the current delegation is a "multi-role" delegation? What does that mean?

Reply:
"multi-role" wording was wrong. This was taken out of spec. I thought it meant a role that delegate to more than 1 more role. but it actually refers to something else that is not in 1.x version of the spec and is being considered for 2.x.

I updated the wording from the current version of the spec.